### DISCUSSION TITLE
how can I get the raw request but still let fastapi to do the type validation?

### DISCUSSION THREAD

**Discussion by braindevices** [fastapi/fastapi#12734] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Request, Depends
from pydantic import BaseModel

app = FastAPI()

class MyData(BaseModel):
    name: str
    age: int

@app.post("/items/")
async def create_item(
    validated_data: MyData,  # FastAPI validates this
):
    # I want some magic that can avoid model_dump_json() and directly let me access the raw request
    return validated_data.model_dump_json()
```

### Description

currently I either have to use request: Request to access the request directly without fastapi do the validation for me, or I have to use the pydantic model to generate the doc and let fastapi validate the request, but then I have to re-serialize the stuff back. How can I let fastapi do all the document generation, validation, etc, but then I can still access the request directly?

### Operating System

Linux

### FastAPI Version

0.115.0

### Pydantic Version

2.9.2

### Python Version

3.10

**Comment by luzzodev** [comment, marked as answer]
``` python
from fastapi import FastAPI, Request, Depends
from pydantic import BaseModel

app = FastAPI()

class MyData(BaseModel):
    name: str
    age: int

@app.post("/items/")
async def create_item(
    request: Request,
    validated_data: MyData,  # FastAPI validates this
):
    # I want some magic that can avoid model_dump_json() and directly let me access the raw request
    return validated_data
```

if you need to access to the raw request or response, you can add them to the params of your function and let pydantic and fastapi do the rest.
